shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/repository/DatasetRepository.java

```diff
@@ -38,4 +39,7 @@ public interface DatasetRepository extends PagingAndSortingRepository<Dataset, L
 
 	void deleteByIdIn(List<Long> ids);
 
+	@Query("SELECT d FROM Dataset d INNER JOIN DatasetAcquisition a WHERE a.examination.id = :examId")
+	Iterable<Dataset> findByExaminationId(Long examId);
```

Comment:
I think you can directly do a findByDatasetAcquisitionExaminationId or findByDatasetAcquisition_ExaminationId without @Query annotation.
See https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-property-expressions